app/src/main/java/com/github/swent/echo/data/supabase/SupabaseDataSource.kt

```diff
@@ -0,0 +1,98 @@
+package com.github.swent.echo.data.supabase
+
+import com.github.swent.echo.data.model.Association
+import com.github.swent.echo.data.model.Event
+import com.github.swent.echo.data.model.Tag
+import com.github.swent.echo.data.model.UserProfile
+import com.github.swent.echo.data.repository.datasources.RemoteDataSource
+import com.github.swent.echo.data.supabase.entities.EventSupabase
+import com.github.swent.echo.data.supabase.entities.EventSupabaseSetter
+import com.github.swent.echo.data.supabase.entities.EventTagSupabase
+import com.github.swent.echo.data.supabase.entities.UserProfileSupabase
+import com.github.swent.echo.data.supabase.entities.UserProfileSupabaseSetter
+import com.github.swent.echo.data.supabase.entities.UserTagSupabase
+import io.github.jan.supabase.SupabaseClient
+import io.github.jan.supabase.postgrest.from
+import io.github.jan.supabase.postgrest.query.Columns
+
+class SupabaseDataSource(supabaseClient: SupabaseClient) : RemoteDataSource {
+
+    var supabase = supabaseClient
+
+    override suspend fun getAssociation(associationId: String): Association {
+        return supabase
+            .from("associations")
+            .select() { filter { eq("association_id", associationId) } }
+            .decodeSingle()
+    }
+
+    override suspend fun getAllAssociations(): List<Association> {
+        return supabase.from("associations").select().decodeList<Association>()
+    }
+
+    override suspend fun getEvent(eventId: String): Event {
+        var event =
+            supabase
+                .from("events")
+                .select(
+                    Columns.raw(
+                        "event_id, user_profiles!public_events_creator_id_fkey(user_id, name), associations(association_id, name, description), title, description, event_tags(tags(tag_id, name)), location_name, location_lat, location_long, start_date, end_date, participant_count, max_participants, image_id"
```

Comment:
Isn't the id of the parent tag missing? Or maybe I can just not read SQL queries...

Reply:
Also, I forgot to add: this should be a constant as you are using it twice. To add constant in Kotlin, you can use `companion object`s:
```kotlin
companion object {
  const val QUERY = "...."
}
```
Just add the `companion object` inside your class and use `QUERY` as constant.

Reply:
You're absolutely right that `parent_tag` field is missing; great catch! This was actually changed while I've already been working on it and I didn't realize it while rebasing. Since the fields is nullable I also didn't catch this during testing. I'll add the `parent_tag`.

And yes, that makes sense to put that query into a constant. Thanks for indicating how to do this :)